prov/sockets/src/sock_ep_dgram.c

```diff
@@ -215,14 +215,21 @@ int sock_dgram_fi_info(void *src_addr, void *dest_addr, struct fi_info *hints,
 
 	if (hints && hints->rx_attr) {
 		(*info)->rx_attr->op_flags |= hints->rx_attr->op_flags;
+		if(hints->rx_attr->caps)
+			(*info)->rx_attr->caps = hints->rx_attr->caps;
```

Comment:
Please leave a space between the if and the open paren: if (...)

Only the primary caps cannot be added.  It is okay to add secondary caps.  I think the defines for SOCK_EP_XX_CAP may need to be broken into PRI_CAP and SEC_CAP